spec/models/ability/guest_spec.rb

```diff
@@ -15,6 +15,8 @@
 
     it_behaves_like 'has access to public features'
 
+    it { expect(subject).not_to be_able_to(:read, other_user.email) }
```

Comment:
Heads up I believe this should be `be_able_to(:read_email, other_user)`. Same below.

Reply:
I _thought_ this is testing the desired outcome of the `read_email` rule? No?

Reply:
I'm not experienced with CanCan, but looking at how the rule is defined, using `read_email` instead of `read`, I would assume you need to use `read_email` here as well. Also, the rule definition does not receive an email, right? It receives a User.

But it should be testable. I made these changes and the spec fails:

<img width="552" alt="screen shot 2018-05-28 at 13 24 58" src="https://user-images.githubusercontent.com/31735/40614402-a81b2f3c-627a-11e8-9cbb-b27729dbd801.png">

Using this makes it pass:

```ruby
it { expect(subject).to be_able_to(:read_email, other_user) }
```

Reply:
You are right. I remember what I did here. The thing is, the read_email rule is not yet renewed yet. So it is hard to test it (and use it in the community view) the right way. 
The one thing I wanted to guard is that a guest user can not read  emails. I am 98%  sure that the test is testing the right thing here. 😇 
See my general comment for a proposal on how to handle this.

Reply:
Still think this is a false positive @F3PiX 😟 I wasn't able to make `be_able_to(:read, other_user.email)` return true even for cases where it should return true, as I showed above.

This is the same error you did initially, where you were testing things in the view as `can?(:read, user.email)` and it didn't work and we changed it to `can?(:read_email, user)`. It makes sense that the same logic would apply here in the expectation.

Reply:
O yes. So sorry. You are right.